scenarios/csharp/dotnet/captioning/Program.cs

```diff
@@ -0,0 +1,417 @@
+        readonly public string[]? languageIDLanguages;
+        readonly public string? inputFile;
+        readonly public string? outputFile;
+        readonly public string? phraseList;
+        readonly public bool suppressOutputEnabled = false;
+        readonly public bool partialResultsEnabled = false;
+        readonly public string? stablePartialResultThreshold;
+        readonly public bool srtEnabled = false;
+        readonly public bool trueTextEnabled = false;
+        readonly public string subscriptionKey;
+        readonly public string region;
+        
+        public UserConfig (
+            bool profanityFilterRemoveEnabled,
+            bool profanityFilterMaskEnabled,
+            string[]? languageIDLanguages,
+            string? inputFile,
+            string? outputFile,
+            string? phraseList,
+            bool suppressOutputEnabled,
+            bool partialResultsEnabled,
+            string? stablePartialResultThreshold,
+            bool srtEnabled,
+            bool trueTextEnabled,
+            string subscriptionKey,
+            string region
+            )
+        {
+            this.profanityFilterRemoveEnabled = profanityFilterRemoveEnabled;
+            this.profanityFilterMaskEnabled = profanityFilterMaskEnabled;
+            this.languageIDLanguages = languageIDLanguages;
+            this.inputFile = inputFile;
+            this.outputFile = outputFile;
+            this.phraseList = phraseList;
+            this.suppressOutputEnabled = suppressOutputEnabled;
+            this.partialResultsEnabled = partialResultsEnabled;
+            this.stablePartialResultThreshold = stablePartialResultThreshold;
+            this.srtEnabled = srtEnabled;
+            this.trueTextEnabled = trueTextEnabled;
+            this.subscriptionKey = subscriptionKey;
+            this.region = region;
+        }
+    }
+
+    public class TooFewArguments : Exception
+    {
+        public TooFewArguments() : base() { }
+    }
+
+    class Program
+    {
+
+/*
+Helper functions
+*/
+
+        static string? GetCmdOption(string[] args, string option)
+        {
+            int index = Array.IndexOf(args, option);
+            if (index > -1 && index < args.Length - 2)
+            {
+// We found the option (for example, "-o"), so advance from that to the value (for example, "filename").
+                return args[index + 1];
+            }
+            else {
+                return null;
+            }
+        }
+
+        static bool CmdOptionExists(string[] args, string option)
+        {
+            return args.Contains (option);
+        }
+
+        static string V2EndpointFromRegion(string region)
+        {
+// Note: Continuous language identification is supported only with v2 endpoints.
+            return "wss://" + region + ".stt.speech.microsoft.com/speech/universal/v2";
+        }
+
+        static string TimestampFromSpeechRecognitionResult(SpeechRecognitionResult result, UserConfig userConfig)
```

Comment:
Given the ubiquity and lack of "varying," this likely shouldn't be a variable and should instead be a member or even shared singleton